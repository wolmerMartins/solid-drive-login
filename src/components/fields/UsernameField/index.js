import React from 'react'

import FieldBase from '../FieldBase'
import InputText from '../../inputs/InputText'

const UsernameField = ({ value, onChange }) => (
  <FieldBase
    label="Usuário"
    id="username-field"
  >
    <InputText
      value={value}
      onChange={onChange}
      id="username-field"
      placeholder="Nome de usuário ou e-mail"
    />
  </FieldBase>
)

export default UsernameField
