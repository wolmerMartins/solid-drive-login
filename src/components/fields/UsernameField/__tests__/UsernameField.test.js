import React, { useState } from 'react'
import { render, fireEvent } from '@testing-library/react'

import UsernameField from '../index'

const TestComponent = () => {
  const [value, setValue] = useState('')

  const onChange = e => setValue(e.target.value)

  return <UsernameField onChange={onChange} value={value} />
}

const buildComponent = () => {
  const utils = render(<TestComponent />)
  const input = utils.getByLabelText('Usu√°rio')

  return {
    input,
    ...utils
  }
}

describe('<UsernameField />', () => {
  it('Render the component without crashing', () => {
    const { asFragment } = buildComponent()

    expect(asFragment())
      .toMatchSnapshot()
  })

  it('Types a value in the input', () => {
    const { input } = buildComponent()

    expect(input)
      .toHaveProperty('value', '')

    fireEvent.change(input, { target: { value: 'username' } })

    expect(input)
      .toHaveProperty('value', 'username')
  })
})
