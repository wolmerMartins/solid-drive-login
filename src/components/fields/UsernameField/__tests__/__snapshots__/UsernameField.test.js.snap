// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsernameField /> Render the component without crashing 1`] = `
<DocumentFragment>
  <fieldset
    class="FieldBase"
  >
    <label
      class="FieldBase-label"
      for="username-field"
    >
      Usuário
    </label>
    <input
      class="InputBase "
      id="username-field"
      placeholder="Nome de usuário ou e-mail"
      type="text"
      value=""
    />
  </fieldset>
</DocumentFragment>
`;
